lib/rubrowser/parser/definition/base.rb

```diff
@@ -23,8 +24,16 @@ def kernel?
           namespace.empty?
         end
 
+        def circular?
+          @circular
+        end
+
+        def set_circular
+          @circular = true
+        end
+
         def ==(other)
-          namespace == other.namespace
+          namespace == other.namespace && circular? == other.circular?
```

Comment:
I did this to facilitate testing. It allowed me to create a double definition and then simply ask if it's equal to the result rather than checking each individual property
```
definition = double(:definition, namespace: 'Some Namespace', circular?: false)
expect(result).to eq(definition)

# vs 

expect(result.namespace).to eq('Some Namespace')
expect(result.circular?).to eq(true)
```

If you still feel strongly, I'll make the change.

Reply:
I thought of this for some time now, here is a situation (maybe an edge case):
1- you have 2 directories, a class is circular in the first one, and not circular in the second one
2- you analyze both directories
3- when you try to join results these 2 objects for the same class will be considered different, but it should be the same.
hummm, I feel strongly against this line sorry :smiley: 
Also sorry for late reply I was getting married this month :smiley:

Reply:
Oh man. Congratulations!

No worries at all. That is a fair point. I'll make the change shortly.